stl/inc/flat_set

```diff
@@ -70,17 +65,17 @@ public:
     static_assert(random_access_iterator<iterator>, "The C++ Standard forbids containers without random "
                                                     "access iterators from being adapted. See [flatset.overview].");

```

Comment:
`See [flatset.overview]` is slightly problematic, as this is also for `flat_multiset`; not dealt with in this pr. (Is it ok to remove these references?)